<h1>Data Analytics </h1>
<hr>
<h3>Name: Roderick L. Alabado</h3>

In [1]:
import numpy as  np
import pandas as pd

<p>Import Dataset</p>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

penguins


<hr>
<h3>Descriptive Analytics</h3>

In [3]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [4]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

In [5]:
penguins.value_counts()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  sex     year
Gentoo   Biscoe  59.6            17.0           230.0              6050.0       male    2007    1
Adelie   Biscoe  34.5            18.1           187.0              2900.0       female  2008    1
                 35.0            17.9           190.0              3450.0       female  2008    1
                                                192.0              3725.0       female  2009    1
                 35.3            18.9           187.0              3800.0       female  2007    1
                                                                                               ..
                 37.9            18.6           193.0              2925.0       female  2009    1
                 38.1            16.5           198.0              3825.0       female  2009    1
                                 17.0           181.0              3175.0       female  2009    1
                 38.2    

In [6]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<hr>
<h2>Selecting Columns</h2>

In [7]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [8]:
penguins['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [9]:
adelie = penguins[penguins['species'] == "Adelie"]

In [10]:
adelie

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [11]:
gentoo = penguins[penguins['species'] == "Gentoo"]

In [12]:
gentoo


species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [13]:
chinstrap = penguins[penguins['species'] == "Chinstrap"]

In [14]:
chinstrap

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [15]:
penguins['body_mass_g'] >= 4000

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Name: body_mass_g, Length: 344, dtype: bool

In [16]:
(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Length: 344, dtype: bool

In [17]:
penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [18]:
chinsttrap_heavy = penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000) ]

In [19]:
chinsttrap_heavy.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


In [20]:
gender = [penguins['sex'] == "male"]

In [21]:
gender

[0       True
 1      False
 2      False
 3      False
 4      False
        ...  
 339     True
 340    False
 341     True
 342     True
 343    False
 Name: sex, Length: 344, dtype: bool]

<hr>
<h2>How to Fill up Null Values</h2>

In [22]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [23]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [24]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [25]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [26]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [27]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [28]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [29]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [30]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

<hr>
<h2>How to Change data Types of Columns</h2>

In [31]:
penguins['sex'] = penguins['sex'].astype('category')
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['year'] = penguins['year'].astype('category')


In [32]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<hr>
<h2>How to create Columns</h2>

In [33]:
penguins['mass_kg'] = penguins['body_mass_g'] / 1000

In [34]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000,1)

In [35]:
penguins


species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

<hr>
<h1>Explanatory Data Analytics(EDA)</h1>

<h2>1. Which sex of penguins is more common?</h2>




In [36]:
penguins['sex'].mode()

0    male
Name: sex, dtype: category
Categories (2, object): ['female', 'male']

In [37]:
penguins['sex'].value_counts()

sex
male      179
female    165
Name: count, dtype: int64

<h3>Insight#1</h3>
<p>Male penguins are slightly more common than females.</p>




<hr>
<h2>2. Which island has more penguins?</h2>

In [38]:
penguins['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h3>Insight#2</h3>
<p>Biscoe has more penguins than the other islands.</p>

<hr>
<h2>3. Which species has the longest average flipper length?</h2>

In [39]:
penguins[penguins['species'] == 'Adelie']['flipper_length_mm'].mean()

np.float64(190.02575792551556)

In [40]:
penguins[penguins['species'] == 'Chinstrap']['flipper_length_mm'].mean()

np.float64(195.8235294117647)

In [41]:
penguins[penguins['species'] == 'Gentoo']['flipper_length_mm'].mean()

np.float64(217.05576777966422)

<h3>Insight#3</h3>
<p>Gentoo penguins have the longest average flipper length. </p>

<hr>
<h2>4. What is the common bill depth among all penguins?</h2>

In [42]:
cmmn_bill_depth = penguins['bill_depth_mm'].mode()

In [43]:
cmmn_bill_depth

0    17.0
Name: bill_depth_mm, dtype: float64

In [44]:
cmmn_bill_depth = penguins['bill_depth_mm'].mode()[0]

In [45]:
cmmn_bill_depth

np.float64(17.0)

<h3>Insight#4</h3>
<p>The common bill depth that most penguins have is 17.0 mm</p>

<hr>
<h2>5. What is the average bill length by gender?</h2>

In [46]:
penguins[penguins['sex'] == 'male']['bill_length_mm'].mean()

np.float64(45.60415564049789)

In [47]:
penguins[penguins['sex'] == 'female']['bill_length_mm'].mean()

np.float64(42.096969696969694)

<h3>Inight#5</h3>
<p>Male penguins generally have longer bills than females.</p>

<hr>
<h2>6. What was the number of penguins measured on each island in 2007?</h2>

In [48]:
island_counts_2007 = penguins[penguins['year'] == 2007]['island']

In [49]:
island_counts_2007 

0      Torgersen
1      Torgersen
2      Torgersen
3      Torgersen
4      Torgersen
         ...    
297        Dream
298        Dream
299        Dream
300        Dream
301        Dream
Name: island, Length: 110, dtype: category
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

In [50]:
island_counts_2007.value_counts()

island
Dream        46
Biscoe       44
Torgersen    20
Name: count, dtype: int64

<h3>Insight #6</h3>
<p>In 2007, Dream Island had the most penguins measured (46), followed closely by Biscoe (44), while Torgersen had the fewest (20). </p>

<hr>
<h2>7. What is the population count for each penguin species?
</h2>

In [51]:
species_counts = penguins['species'].value_counts()

In [52]:
species_counts

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h3>Insight #7</h3>
<p>The results shows that Adelie has the most population count with 152, Gentoo comes next with 124, and Chinstrap has 68.</p>

<h2>8. Which species has the widest average bill?
</h2>

In [53]:
bill_depth = penguins.groupby('species')['bill_depth_mm'].mean()

C:\Users\63968\AppData\Local\Temp\ipykernel_12508\239996347.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  bill_depth = penguins.groupby('species')['bill_depth_mm'].mean()


In [54]:
bill_depth

species
Adelie       18.338495
Chinstrap    18.420588
Gentoo       14.999606
Name: bill_depth_mm, dtype: float64

<h3>Insight#8</h3>
    
 <p>Chinstrap penguins have the widest average bill depth among the three species.</p>

<h2>9. Which island has the highest average flipper length?
</h2>

In [55]:
island_flipper = penguins.groupby('island')['flipper_length_mm'].mean()

C:\Users\63968\AppData\Local\Temp\ipykernel_12508\168910784.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  island_flipper = penguins.groupby('island')['flipper_length_mm'].mean()


In [56]:
island_flipper

island
Biscoe       209.654257
Dream        193.072581
Torgersen    191.382985
Name: flipper_length_mm, dtype: float64

<h3>Insight#9</h3>
    
 <p>Biscoe island have the highest average flipper lenght among the three island.</p>

<h2>10. Which species has the largest average body mass?
</h2>

In [57]:
body_mass = penguins.groupby('species')['body_mass_g'].mean()


C:\Users\63968\AppData\Local\Temp\ipykernel_12508\2725598840.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  body_mass = penguins.groupby('species')['body_mass_g'].mean()


In [58]:
body_mass

species
Adelie       3703.958910
Chinstrap    3733.088235
Gentoo       5068.965761
Name: body_mass_g, dtype: float64

<h3>Insight#10</h3>
    
 <p>Gentoo penguins have the largest average body mass compared to Adelie and Chinstrap.</p>

<h2>11. What is the average bill depth by island and sex?
</h2>

In [59]:
avg_dpt = penguins.groupby(['island','sex'])['bill_depth_mm'].mean()

C:\Users\63968\AppData\Local\Temp\ipykernel_12508\1341882794.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  avg_dpt = penguins.groupby(['island','sex'])['bill_depth_mm'].mean()


In [60]:
avg_dpt

island     sex   
Biscoe     female    15.191250
           male      16.510809
Dream      female    17.601639
           male      19.063492
Torgersen  female    17.550000
           male      19.137542
Name: bill_depth_mm, dtype: float64

<h3>Insight#11</h3>
    
 <p>Across all islands, male penguins consistently have higher bill depth average than females. The difference is most noticeable on Dream and Torgersen Islands, where males’ bill depth are about 1.5 mm higher on average.</p>

<h2>12. Which island has the smallest average bill length?
</h2>

In [61]:
bill_lgth = penguins.groupby('island')['bill_length_mm'].mean()

C:\Users\63968\AppData\Local\Temp\ipykernel_12508\1642300992.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  bill_lgth = penguins.groupby('island')['bill_length_mm'].mean()


In [62]:
bill_lgth

island
Biscoe       45.249535
Dream        44.167742
Torgersen    39.046576
Name: bill_length_mm, dtype: float64

<h3>Insight#12</h3>
    
 <p>Torgersen Island penguins have the smallest bill length on average.</p>